azureml/pipeline.py

```diff
@@ -0,0 +1,198 @@
+import os
+import sys
+import argparse
+
+from azure.ai.ml import MLClient, Input, Output, dsl, command
+from azure.ai.ml.parallel import parallel_run_function, RunFunction
+import azure.ai.ml.entities
+
+project_dir = os.path.abspath(os.path.join(os.path.dirname(__file__), ".."))
+sys.path.append(project_dir)
+
+from rag_experiment_accelerator.config.environment import Environment  # noqa: E402
+from rag_experiment_accelerator.config.config import Config  # noqa: E402
+from rag_experiment_accelerator.config.index_config import IndexConfig  # noqa: E402
+from rag_experiment_accelerator.config.paths import mlflow_run_name  # noqa: E402
+from rag_experiment_accelerator.utils.auth import get_default_az_cred  # noqa: E402
+
+
+AML_ENVIRONMENT_NAME = "rag-env"
+INDEX_STEP_RETRIES = 3
+INDEX_STEP_TIMEOUT_SECONDS = 10 * 3600
+INDEX_STEP_FILES_PER_BATCH = 4
+INDEX_STEP_CONCURRENCY_PER_INSTANCE = 4
```

Comment:
This has to be provided for the parallel job, which honestly is not great. I will add a ticket here to improve this